### PR TITLE
Solution space/path finder improving

### PR DESCRIPTION
**Pull request by elena-lyulina**
This PR includes a lot of small improvements such as:
1. Creating empty vertex in the graph
2. Add ability to find rate during Code creation
3. Change traversal a little to be able to include/not include start and end vertices
4. Change test_system so now it's possible not to print equal docs
5. Add multithreading to distance and remove it from graph, since we don't use it anymore

### CODE REVIEW

File: src/main/solution_space/distance.py [hunk 1]
```diff
@@ -1,44 +1,88 @@
 # Copyright (c) 2020 Anastasiia Birillo, Elena Lyulina
```
Comment (nbirillo) [inline src/main/solution_space/distance.py:1]: Please, write **Todo comment** with my comments in the file, we will fix it in the future because now we don't use distance

File: src/main/solution_space/path_finder/path_finder.py [hunk 2]
```diff
@@ -4,13 +4,10 @@
 
 import logging
 from abc import ABCMeta, abstractmethod
+from typing import Type, Optional
```
Comment (nbirillo) [inline src/main/solution_space/path_finder/path_finder.py:7]: Import's order

File: src/main/solution_space/path_finder_test_system.py [hunk 3]
```diff
@@ -69,18 +71,25 @@ def __init__(self, test_inputs: List[TestInput], serialized_graph_path: str = TE
...
+        methods_dict, methods_names = self.__get_methods_dict_and_names(classes, methods_to_keep)
+        table = PrettyTable(title=title)
+        table.add_column('class_name', [c.__name__ for c in classes])
+        for name in methods_names:
+            methods = [methods_dict[c].get(name) for c in classes]
+            docs = [self._no_method_sign if m is None else self.__format_doc_str(m.__doc__) for m in methods]
+            if not self._add_same_docs and len(set(docs)) == 1:
+                continue
+            table.add_column(name, docs)
+        # for c in classes:
```
Comment (nbirillo) [inline src/main/solution_space/path_finder_test_system.py:85]: Should we delete the comment?

File: src/main/util/consts.py [hunk 4]
```diff
@@ -192,7 +192,7 @@ class SPLIT(Enum):
     TASKS = 'tasks'
 
 
-LOGGER_FILE = ROOT_DIR + '../../../../logs.log'
+LOGGER_FILE = ROOT_DIR + '../../../../logs_1.log'
```
Comment (nbirillo) [inline src/main/util/consts.py:195]: Why did you rename it?

File: src/test/solution_space/distance/multithreaded_filling_dist_test.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2020 Anastasiia Birillo, Elena Lyulina
+import logging
```
Comment (nbirillo) [inline src/test/solution_space/distance/multithreaded_filling_dist_test.py:2]: Please, add an empty line

File: src/main/solution_space/distance.py [hunk 6]
```diff
@@ -1,44 +1,88 @@
...
-            return False
-
         if new_item in self._dist.keys():
             log.info('This item already exists')
             return False
+
+        # If we don't store dist, we don't fill self._dist[new_item],
+        # but we add an empty dict to be able to fill self._dist later
         self._dist[new_item] = {}
+        if not self._to_store_dist:
```
[the remark this answers is not in the document]
Comment (elena-lyulina) [reply src/main/solution_space/distance.py:82]: No, we shouldn't. We need to store an empty dict for each vertex because when we will fill dist_matrix we should have all vertices as keys. 
If we swap these lines, we won't be able to fill dist matrix.
I wrote a comment about it right above line 81.

File: src/main/solution_space/path_finder_test_system.py [hunk 7]
```diff
@@ -193,4 +200,16 @@ def __get_all_subclasses(cls, clazz: Type[Class]) -> List[Type[Class]]:
...
+        not_skipped_subclasses = subclasses.copy()
+        for c in subclasses:
+            if hasattr(c, 'is_skipped') and c.is_skipped:
+                not_skipped_subclasses.remove(c)
+                TestSystem.__print_output(f'{c.__name__} is skipped, reason: {c.skipped_reason}')
+        return not_skipped_subclasses
+
+    # Todo: add ability to print output to file?
+    @staticmethod
+    def __print_output(output: Any) -> None:
```
Comment (nbirillo) [inline src/main/solution_space/path_finder_test_system.py:213]: Please, use `Optional[Any]` for the `output ` param
